src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/AbstractQuteTemplateLinkCollector.java

```diff
@@ -37,228 +38,250 @@
-			if (templateFile != null) {
-				return templateFile;
-			}
-		}
-		return getVirtualFile(project, templateFilePathWithoutExtension, PREFERRED_SUFFIX);
-	}
+    @Nullable
+    private VirtualFile getVirtualFile(VirtualFile resourcesDir, String templateFilePathWithoutExtension, String suffix) {
+        if (resourcesDir != null) {
+            VirtualFile templatesDir = resourcesDir.findFileByRelativePath(TEMPLATES_FOLDER_NAME);
+            if (templatesDir != null && templatesDir.exists()) {
+                VirtualFile templateFile = templatesDir.findFileByRelativePath(templateFilePathWithoutExtension + suffix);
+                if (templateFile != null && templateFile.exists()) {
+                    return templateFile;
+                }
+            }
+        }
+        return null;
+    }
 
-	@Nullable
-	private VirtualFile getVirtualFile(Module project, String templateFilePathWithoutExtension, String suffix) {
-		for(VirtualFile root : ModuleRootManager.getInstance(project).getContentRoots()) {
-			VirtualFile templateFile = root.findFileByRelativePath(templateFilePathWithoutExtension + suffix);
-			if (templateFile != null && templateFile.exists()) {
-				return templateFile;
-			}
-		}
-		return null;
-	}
+    protected String getVirtualFileUrl(Module project, String templateFilePathWithoutExtension, String suffix) {
+        try {
+            for (VirtualFile root : ModuleRootManager.getInstance(project).getContentRoots()) {
+                return new URL(LSPIJUtils.toUri(root).toURL(), templateFilePathWithoutExtension + suffix).toString();
+            }
+        } catch (MalformedURLException e) {
+        }
+        return null;
+    }
 
-	protected String getVirtualFileUrl(Module project, String templateFilePathWithoutExtension, String suffix) {
-		try {
-			for(VirtualFile root : ModuleRootManager.getInstance(project).getContentRoots()) {
-				return new URL(LSPIJUtils.toUri(root).toURL(), templateFilePathWithoutExtension + suffix).toString();
-			}
-		} catch (MalformedURLException e) {}
-		return null;
-	}
+    /**
+     * Returns the relative template file path from the module.
+     *
+     * @param templateFilePath the template file path (without file extension).
+     * @param templateFile     the template file and null otherwise.
+     * @param module           the module.
+     * @return the relative template file path from the module.
+     */
+    private @NotNull String getRelativePath(@NotNull String templateFilePath, @Nullable VirtualFile templateFile, @NotNull Module module) {
+        if (templateFile != null) {
+            // The template file exists
+            // Compute the relative path from the module root path:
+            // - src/main/resources/templates/page.html for Maven project
+            // - resources/templates/page.html for Gradle project
+            for (VirtualFile root : ModuleRootManager.getInstance(module).getContentRoots()) {
```

Comment:
You'll likely need to use this trick
https://github.com/redhat-developer/intellij-quarkus/pull/1240/files#diff-75cbc8b0d9b6203597c60ac5fb2193eccbd1308d712e1a880c67d078b21d7889R213

to keep non-generated source folders first

Reply:
fixed